contrib/adapters/ale.py

```diff
@@ -0,0 +1,214 @@
+#
+# Copyright 2017 Pixar Animation Studios
+#
+# Licensed under the Apache License, Version 2.0 (the "Apache License")
+# with the following modification; you may not use this file except in
+# compliance with the Apache License and the following modification to it:
+# Section 6. Trademarks. is deleted and replaced with:
+#
+# 6. Trademarks. This License does not grant permission to use the trade
+#    names, trademarks, service marks, or product names of the Licensor
+#    and its affiliates, except as required to comply with Section 4(c) of
+#    the License and to reproduce the content of the NOTICE file.
+#
+# You may obtain a copy of the Apache License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the Apache License with the above modification is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied. See the Apache License for the specific
+# language governing permissions and limitations under the Apache License.
+#
+
+"""OpenTimelineIO Avid Log Exchange (ALE) Adapter"""
+
+import opentimelineio as otio
+
+
+class ALEParseError(otio.exceptions.OTIOError):
+    pass
+
+
+def read_from_string(input_str, fps=24):
+
+    collection = otio.schema.SerializableCollection()
+    header = {}
+    columns = []
+
+    lines = input_str.split("\n")
+    while len(lines):
+        line = lines.pop(0)
+
+        # skip blank lines
+        if line.strip() == "":
+            continue
+
+        if line.strip() == "Heading":
+            while len(lines):
+                line = lines.pop(0)
+
+                if line.strip() == "":
+                    break
+
+                if "\t" not in line:
+                    raise ALEParseError("Invalid Heading line: "+line)
+
+                line = line.strip('\r')
+                segments = line.split("\t")
+                while len(segments) >= 2:
+                    key, val = segments.pop(0), segments.pop(0)
+                    header[key] = val
+                if len(segments) != 0:
+                    raise ALEParseError("Invalid Heading line: "+line)
+
+        if "FPS" in header:
+            fps = float(header["FPS"])
+
+        if line.strip() == "Column":
+            if len(lines) == 0:
+                raise ALEParseError("Unexpected end of file after: "+line)
+
+            line = lines.pop(0)
+            columns = line.split("\t")
+
+        if line.strip() == "Data":
+            while len(lines):
+                line = lines.pop(0)
+
+                if line.strip() == "":
+                    continue
+
+                row = line.split("\t")
+
+                if len(row) < len(columns):
+                    raise ALEParseError("Too few values on row: "+line)
```

Comment:
There are some ALEs that rather than writing back-to-back delimiters will just prematurely end the line. I suggest we fill in with empty strings in this case:

`columns.extend([""] * (len(row) - len(columns)))`

Reply:
Will do.